pkg/connector/twitterfmt/convert.go

```diff
@@ -0,0 +1,130 @@
+package twitterfmt
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"maunium.net/go/mautrix/bridgev2"
+	"maunium.net/go/mautrix/bridgev2/networkid"
+	"maunium.net/go/mautrix/event"
+
+	"go.mau.fi/mautrix-twitter/pkg/twittermeow/data/types"
+)
+
+func Parse(ctx context.Context, portal *bridgev2.Portal, msg *types.MessageData) *event.MessageEventContent {
+	bodyHtml := strings.Builder{}
+	charArr := []rune(msg.Text)
+	cursor := 0
+	sortedEntites := sortEntities(msg.Entities)
+
+	for _, entity := range sortedEntites {
+		if entity.URL != nil {
+			url := entity.URL
+			start, end := url.Indices[0], url.Indices[1]
+			if cursor < start {
+				bodyHtml.WriteString(string(charArr[cursor:start]))
+			}
+			bodyHtml.WriteString(url.ExpandedURL)
+			cursor = end
+		}
+		if entity.UserMention != nil {
+			mention := entity.UserMention
+			start, end := mention.Indices[0], mention.Indices[1]
+			if cursor < start {
+				bodyHtml.WriteString(string(charArr[cursor:start]))
+			}
+
+			uid := mention.IDStr
+			dmPortals, err := portal.Bridge.GetDMPortalsWith(ctx, networkid.UserID(uid))
+			if err != nil {
+				continue
+			}
+
+			if len(dmPortals) != 0 {
+				fmt.Fprintf(&bodyHtml,
+					`<a href="%s">%s</a>`,
+					dmPortals[0].MXID.URI().MatrixToURL(),
+					dmPortals[0].Name,
+				)
+			} else {
+				userLogin := portal.Bridge.GetCachedUserLoginByID(networkid.UserLoginID(uid))
+				text := "@" + mention.ScreenName
+				if userLogin != nil {
+					fmt.Fprintf(&bodyHtml,
+						`<a href="%s">%s</a>`,
+						userLogin.UserMXID.URI().MatrixToURL(),
+						text,
+					)
+
+				} else {
+					bodyHtml.WriteString(text)
+				}
+			}
```

Comment:
This part seems wrong, it should just always mention the ghost and never portals. Even other user logins are probably not worth looking up.

Reply:
Handling in old bridge for reference: https://github.com/mautrix/twitter/blob/v0.1.8/mautrix_twitter/formatter.py#L31-L56